src/plugins/oSnap/components/Input/Amount.vue

```diff
@@ -0,0 +1,47 @@
+<script setup lang="ts">
+import { formatUnits, parseUnits } from '@ethersproject/units';
+
+const props = defineProps<{
+  modelValue: string;
+  label: string;
+  decimals: number | undefined;
+}>();
+const emit = defineEmits<{
+  'update:modelValue': [value: string];
+}>();
+
+const input = ref('0');
+const isValid = ref(true);
+const dirty = ref(false);
+
+const format = (amount: string) => {
+  try {
+    return parseUnits(amount, props.decimals).toString();
+  } catch (error) {
+    return undefined;
+  }
+};
+
+const handleInput = () => {
+  dirty.value = true;
+  const value = format(input.value);
+  isValid.value = !!value;
+  emit('update:modelValue', value ?? '');
+};
+
+onMounted(() => {
+  if (props.modelValue) {
+    input.value = formatUnits(props.modelValue, props.decimals);
+  }
+});
+</script>
+
+<template>
+  <UiInput
```

Comment:
Would be great if you could use Tune components instead of `/Ui` as we are in the process of deprecating these. But if this will create major difficulties for you(e.g making many changes to Tune) then I;d say we can refactor at a later time.